jaxtronomy/LensModel/Solver/lens_equation_solver.py

```diff
@@ -0,0 +1,773 @@
+        specialized recipe that solves a one-dimensional lens equation that is easier
+        and more reliable to solve than the usual two-dimensional lens equation.
+
+        :param x: source position in units of angle, an array of positions is also supported.
+        :param y: source position in units of angle, an array of positions is also supported.
+        :param kwargs_lens: lens model parameters as keyword arguments
+        :param arrival_time_sort: bool, if True, sorts image position in arrival time (first arrival photon first listed)
+        :param magnification_limit: None or float, if set will only return image positions that have an
+         abs(magnification) larger than this number
+        :param kwargs_solver: additional kwargs to be supplied to the solver. Particularly relevant are Nmeas and Nmeas_extra
+        :returns: (exact) angular position of (multiple) images ra_pos, dec_pos in units of angle
+         Note: in contrast to the other solvers, generally the (heavily demagnified) central image will also be included, so
+         setting a proper magnification_limit is more important. To get similar behaviour, a limit of 1e-1 is acceptable
+        """
+        lens_model_list = copy.deepcopy(list(self.lensModel.lens_model_list))
+
+        # make MST when "CONVERGENCE" profile is given
+        if "CONVERGENCE" in lens_model_list:
+            # here we apply an inverse MST that leaves image positions invariant under the MST
+            kwargs_lens_ = copy.deepcopy(kwargs_lens)
+            index_convergence = lens_model_list.index("CONVERGENCE")
+
+            # MST in source position and Einstein radius
+            kappa = kwargs_lens_[index_convergence]["kappa"]
+            ra0 = kwargs_lens_[index_convergence].get("ra_0", 0)
+            dec0 = kwargs_lens_[index_convergence].get("dec_0", 0)
+            lambda_mst = (
+                1 - kappa
+            )  # a mass sheet that compensates the convergence field
+            # source position mapping
+            x_ = (x - ra0) / lambda_mst
+            y_ = (y - dec0) / lambda_mst
+            # lens mapping
+            # power-law scaling with mst
+            # alpha = theta_E * (r2 / theta_E**2) ** (1 - gamma / 2.0)
+            gamma = kwargs_lens[0]["gamma"] if "gamma" in kwargs_lens[0] else 2
+
+            kwargs_lens_[0]["theta_E"] /= lambda_mst ** (1.0 / (gamma - 1))
+            if "SHEAR" in lens_model_list:
+                kwargs_lens_[1]["gamma1"] /= lambda_mst
+                kwargs_lens_[1]["gamma2"] /= lambda_mst
+
+            # removing of kwargs_lens of "CONVERGENCE" profile
+            kwargs_lens_.pop(index_convergence)
+            lens_model_list.pop(index_convergence)
+        else:
+            kwargs_lens_ = kwargs_lens
+            x_, y_ = x, y
+
+        if lens_model_list not in SUPPORTED_LENS_MODELS_ANALYTICAL:
+            raise ValueError(
```

Comment:
you can do raise testing with numpy.testing assert_raise() and then specify what error should be raised for the input (i.e. ValueError)

Reply:
Ah OK, will implement that